docs/proposals/apiman-integration.md

```diff
@@ -0,0 +1,123 @@
+# APIMan Integration
+
+## Abstract
+This proposes a design for integrating [APIMan](http://www.apiman.io/) with OpenShift in order to provide  micro-services governance for API service providers.
+
+## Motivation
+Providers of services may have the need to control who and how their services are consumed.  This may be realized as policies defined by providers to control a service
+in any number of ways (e.g security, throttling/quota, billing and metrics)<sup>[1](#r1)</sup>. 
+
+## Constraints and Assumptions
+### Authentication
+The OAuth token identifying a user in OpenShift will be utilized by APIMan to associate service policy to their identity.  This will ensure a user's identity is 
+ubiquitous across the cluster when interacting with OpenShift and the APIMan management console.
+
+### Deployment
+#### Communication
+APIMan Components (e.g. gateway, management interface) integrated with the cluster will utilize mutual TLS for communication.
+
+#### On-Cluster Policy Storage
+[Origin Aggregated logging](https://github.com/openshift/origin-aggregated-logging)<sup>[4](#r4)</sup> will be deployed before APIMan in order to utilize the existing ElasticSearch cluster for policy storage.  
+The [ACL plugin](https://github.com/fabric8io/openshift-elasticsearch-plugin)<sup>[2](#r2)</sup> that is deployed
+as part of aggregated logging will be updated to allow access from the gateway.  ElasicSearch index management<sup>[3](#r3)</sup>  will be modified so
+that it will not cull APIMan policy data.
```

Comment:
If we were going down this route we would also have to designate index namespace for APIMan - something beginning with a `.` since just about anything else could be a project. E.g. `.apiman.` plus whatever the default names are.